### ISSUE TITLE
Uncaught Error: Class 'TYPO3\Surf\Command\SelfUpdateCommand' not found

### ISSUE THREAD

**Issue by YetiCGN** [TYPO3/Surf#735]
## Expected Behavior

Surf to run properly

## Actual Behavior

```
PHP Fatal error:  Uncaught Error: Class 'TYPO3\Surf\Command\SelfUpdateCommand' not found in /tmp/_surf/Container3wlWNNX/getConsoleApplicationService.php:25
Stack trace:
#0 /tmp/_surf/Container3wlWNNX/TYPO3_Surf_Cli_Symfony_ConsoleKernelProdContainer.php(155): Container3wlWNNX\getConsoleApplicationService::do()
#1 phar:///mnt/c/myproject/surf.phar/vendor/symfony/dependency-injection/Container.php(238): Container3wlWNNX\TYPO3_Surf_Cli_Symfony_ConsoleKernelProdContainer->load()
#2 phar:///mnt/c/myproject/surf.phar/vendor/symfony/dependency-injection/Container.php(220): Symfony\Component\DependencyInjection\Container->make()
#3 phar:///mnt/c/myproject/surf.phar/bin/surf(22): Symfony\Component\DependencyInjection\Container->get()
#4 /mnt/c/myproject/surf.phar(14): require('phar:///mnt/o/P...')
#5 {main}
  thrown in /tmp/_surf/Container3wlWNNX/getConsoleApplicationService.php on line 25
  ```

## Steps to Reproduce the Problem

  1. Download 3.3.10 Phar (or 3.3.8 or 3.3,9 for that matter)
  1. Run `surf`
  1. See error

## Specifications

  - Surf Version: 3.3.8, 3.3.9 and 3.3.10, both as Phar and project dependency
  - Application: TYPO3
  - PHP Version: 7.4, 8.1 and 8.2 tested
  - Platform: Ubuntu Linux
  - Environment (CI): doesn't matter
  - Deployment configuration: doesn't even get that far

**Comment by YetiCGN** [comment]
I had to run `rm -rf /tmp/_surf` once on my local machine to make it work. I could introduce it into the release workflow at `.github/workflows/release.yaml`, but I'm unsure whether that's the right thing to do.

**Comment by YetiCGN** [comment]
@sabbelasichon It seems the `tmp` directory is causing more harm than good. For example, it doesn't matter which Phar you run. If there's a previous version in `/tmp/_surf`, you won't be able to use another version until you manually clear that dir:

![image](https://user-images.githubusercontent.com/447119/229846467-f1885e1f-3bf3-439a-ab5c-b306cf69578f.png)

**Comment by sabbelasichon** [comment]
@YetiCGN I see. The current cacheDir is hardcoded and not flexible enough. I am going to change it.

**Comment by sabbelasichon** [comment]
@YetiCGN I have released version 3.3.11. Hopefully not breaking again something else. But i have tested it and the directory for building the Symfony Di Container is now projected related (as in Symfony) and it works also in the Phar context.
